internal/service/federatedsettingsidentityprovider/data_source_federated_settings_identity_provider.go

```diff
@@ -281,11 +281,7 @@ func dataSourceMongoDBAtlasFederatedSettingsIdentityProviderRead(ctx context.Con
 		return diag.FromErr(fmt.Errorf("error setting `status` for federatedSettings IdentityProviders: %s", err))
 	}
 
-	if err := d.Set("id", federatedSettingsIdentityProvider.Id); err != nil {
-		return diag.FromErr(fmt.Errorf("error setting `id` for federatedSettings IdentityProviders: %s", err))
-	}
-
-	d.SetId(federatedSettingsIdentityProvider.OktaIdpId)
```

Comment:
I am also not sure about this change.. (sorry I am writing this after my other comments and so it looks like on the opposite order..) why are we removing the `err` check? and why are we "always" fixing the `Id` field? I thought it was based on understanding what Id were customers using